File: EMDRAppleTest1.1/StatusBarView.swift
import SwiftUI

struct StatusBarView: View {
    @ObservedObject var bluetoothManager: BluetoothManager
    @State private var lastBat1Update = Date()
    @State private var lastBat2Update = Date()
    let batteryTimeout: TimeInterval = 10
    let timer = Timer.publish(every: 1, on: .main, in: .common).autoconnect()
    
    var body: some View {
        HStack {
            // Left battery indicator
            HStack(spacing: 4) {
                Circle()
                    .fill(isLeftBatteryMissing ? Color.red : Color.green)
                    .frame(width: 8, height: 8)
                Text("L: \(bluetoothManager.battery1Level)%")
            }
            
            Spacer()
            
            // Connection status
            Text("\(bluetoothManager.connectedDeviceCount) Connected")
                .foregroundColor(bluetoothManager.connectedDeviceCount > 0 ? .green : .red)
            
            Spacer()
            
            // Right battery indicator
            HStack(spacing: 4) {
                Text("R: \(bluetoothManager.battery2Level)%")
                Circle()
                    .fill(isRightBatteryMissing ? Color.red : Color.green)
                    .frame(width: 8, height: 8)
            }
        }
        .font(.caption)
        .padding(.horizontal)
        .onReceive(timer) { _ in
            let now = Date()
            // Update battery status indicators
            isLeftBatteryMissing = now.timeIntervalSince(lastBat1Update) > batteryTimeout
            isRightBatteryMissing = now.timeIntervalSince(lastBat2Update) > batteryTimeout
        }
        .onReceive(NotificationCenter.default.publisher(for: .batteryLevelUpdated)) { notification in
            if let isLeft = notification.userInfo?["isLeft"] as? Bool {
                if isLeft {
                    lastBat1Update = Date()
                } else {
                    lastBat2Update = Date()
                }
            }
        }
    }
    
    // Add state properties for battery status
    @State private var isLeftBatteryMissing = false
    @State private var isRightBatteryMissing = false
} 

File: lib/views/status_bar_view.dart
import 'package:flutter/material.dart';
import 'package:provider/provider.dart';
import '../bluetooth/bluetooth_manager.dart';

class StatusBarView extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    final bluetoothManager = Provider.of<BluetoothManager>(context);
    
    return Padding(
      padding: const EdgeInsets.symmetric(horizontal: 16.0),
      child: Row(
        children: [
          // Left battery indicator
          Row(
            children: [
              Container(
                width: 8,
                height: 8,
                decoration: BoxDecoration(
                  shape: BoxShape.circle,
                  color: bluetoothManager.leftHeartbeatMissing ? Colors.red : Colors.green,
                ),
              ),
              const SizedBox(width: 4),
              Text(
                'L: ${bluetoothManager.battery1Level}%',
                style: Theme.of(context).textTheme.caption,
              ),
            ],
          ),
          
          Spacer(),
          
          // Connection status
          Text(
            '${bluetoothManager.connectedDeviceCount} Connected',
            style: Theme.of(context).textTheme.caption?.copyWith(
              color: bluetoothManager.connectedDeviceCount > 0 ? Colors.green : Colors.red,
            ),
          ),
          
          Spacer(),
          
          // Right battery indicator
          Row(
            children: [
              Text(
                'R: ${bluetoothManager.battery2Level}%',
                style: Theme.of(context).textTheme.caption,
              ),
              const SizedBox(width: 4),
              Container(
                width: 8,
                height: 8,
                decoration: BoxDecoration(
                  shape: BoxShape.circle,
                  color: bluetoothManager.rightHeartbeatMissing ? Colors.red : Colors.green,
                ),
              ),
            ],
          ),
        ],
      ),
    );
  }
} 